panel/src/components/Views/ModelView.vue

```diff
@@ -37,92 +37,93 @@ export default {
 		},
 		uuid: String
 	},
+	data() {
+		return {
+			isSaved: true
+		};
+	},
 	computed: {
 		changes() {
-			return this.$panel.content.changes;
+			return this.$panel.content.changes(this.api);
 		},
 		editor() {
 			return this.lock.user.email;
 		},
+		hasChanges() {
+			return length(this.changes) > 0;
+		},
 		hasTabs() {
 			return this.tabs.length > 1;
 		},
 		isLocked() {
 			return this.lock.isLocked;
 		},
-		isUnsaved() {
-			return this.$panel.content.hasChanges;
-		},
 		modified() {
 			return this.lock.modified;
-		},
-		protectedFields() {
-			return [];
 		}
 	},
 	mounted() {
-		this.autosave = throttle(this.autosave, 1000, {
-			leading: true,
-			trailing: true
-		});
-
-		this.$events.on("model.reload", this.$reload);
+		this.$events.on("beforeunload", this.onBeforeUnload);
+		this.$events.on("content.save", this.onContentSave);
 		this.$events.on("keydown.left", this.toPrev);
 		this.$events.on("keydown.right", this.toNext);
-		this.$events.on("view.save", this.onSave);
+		this.$events.on("model.reload", this.$reload);
+		this.$events.on("view.save", this.onViewSave);
 	},
 	destroyed() {
-		this.$events.off("model.reload", this.$reload);
+		this.$events.off("beforeunload", this.onBeforeUnload);
+		this.$events.off("content.save", this.onContentSave);
 		this.$events.off("keydown.left", this.toPrev);
 		this.$events.off("keydown.right", this.toNext);
-		this.$events.off("view.save", this.onSave);
+		this.$events.off("model.reload", this.$reload);
+		this.$events.off("view.save", this.onViewSave);
 	},
 	methods: {
-		autosave() {
-			if (this.isLocked === true) {
-				return false;
+		onBeforeUnload(e) {
+			if (this.$panel.content.isProcessing === true || this.isSaved === false) {
+				e.preventDefault();
+				e.returnValue = "";
 			}
-
-			this.$panel.content.save();
 		},
-		async onDiscard() {
-			if (this.isLocked === true) {
-				return false;
+		onContentSave({ api }) {
+			if (api === this.api) {
+				this.isSaved = true;
 			}
-
-			await this.$panel.content.discard();
-			this.$panel.view.reload();
+		},
+		async onDiscard() {
+			await this.$panel.content.discard(this.api);
+			this.$panel.view.refresh();
 		},
 		onInput(values) {
-			if (this.isLocked === true) {
-				return false;
-			}
+			// update the content for the current view
+			// this will also refresh the content prop
+			this.$panel.content.update(values, this.api);
+			// trigger a throttled save call with the updated content
+			this.$panel.content.saveLazy(this.content, this.api);
```

Comment:
`Discussion:` This seems a bit unintuitive: Why do we need to call `saveLazy` separately? Shouldn't any call of `update` also trigger a lazy save? Which situation to we have where only an update but no save should happen? I would think that this is actually a situation that we really want to avoid. Then the content module should take care of it itself and not put that responsibility out in the calling code.

Reply:
You might be right that a separate call is not needed at all. We had the seperate update call for a while. That's mainly why I kept it that way. But I also don't think that there are a lot of good reasons for it.

Reply:
I remember again why we need this. We want the content prop to be updated immediately, while saving will happen only a second later. When we put the update part into the save method, it will also be delayed and create weird side-effects.

Reply:
But we wouldn't put the update part into the save method, but rather make the update method do the update logic and then call itself the `saveLazy` method, ensuring that this update eventually will get saved.

Reply:
Ah, right. But would we then also need a update and updateLazy method or would you handle that with a method argument?

Reply:
Why would you want to offer the option? I think every update should trigger saveLazy.

Reply:
All of our api calls are asynchronous and I would call the default expectation for the api calls in the content method as well. We only need the throttled one for our text inputs. But let's say you want to modify a toggle, that toggle action could totally be sent directly instead of waiting. That's why I thought it would be great to have both. The async one and a lazy one

Reply:
I don't know, to me it would feel easier to have just one method (`update`) that developers use and we take care of all of it internally in the module. Not having them think about "do I need update or updateLazy". And I think always lazy would be just fine. Yes modifying a toggle wouldn't need it, but it also doesn't hurt, the delay is a mere second. And we don't mind that delay in our normal form with toggles either, there the delay is also applied no matter if it's text or a toggle.

Reply:
Ok, convinced. I implemented it like that.